dev/bazel/dal.bzl

```diff
@@ -344,22 +343,13 @@ def _test_eternal_datasets_args(framework):
         })
     return []
 
-def _test_nightly_args(framework):
-    if framework == "catch2":
-        return _select({
-            "@config//:test_nightly_enabled": [],
-            "//conditions:default": [
-                "~[nightly]",
-            ],
-        })
-    return []
-
-def _test_weekly_args(framework):
+def _test_filter_args(framework):
     if framework == "catch2":
         return _select({
+            "@config//:test_nightly_enabled": ["~[weekly]"],
             "@config//:test_weekly_enabled": [],
```

Comment:
What is behavior if both options `--test_nightly` and `--test_weekly` are passed?

Reply:
The tests will not run with the error that only one of these flags is needed.

Reply:
I've tried this, error message is not very clear, but probably fine for now